### PR TITLE
fix(resize): Add check to resize volumes triggered via zvol opcode

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>
This PR adds check to perform resize of volumes via zvol opcode.

### CODE REVIEW

**Review by vishnuitta — APPROVED** [review]
changes are good

File: src/mgmt_conn.c [hunk 1]
```diff
@@ -1042,6 +1042,15 @@ uzfs_zvol_execute_async_command(void *arg)
 		break;
 	case ZVOL_OPCODE_RESIZE:
 		volsize = *(uint64_t *)async_task->payload;
+		if (volsize < ZVOL_VOLUME_SIZE(zinfo->main_zv)) {
```
Comment (vishnuitta) [inline src/mgmt_conn.c]: lets push this after lock

**Review by vishnuitta — CHANGES_REQUESTED** [review]
lets push this after lock

File: src/mgmt_conn.c [hunk 2]
```diff
@@ -1588,6 +1588,16 @@ process_message(uzfs_mgmt_conn_t *conn)
 			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED, hdrp);
 			break;
 		}
+		if (resize_data->size < ZVOL_VOLUME_SIZE(zinfo->main_zv)) {
+			LOGERRCONN(conn, "Failed to resize main volume %s, "
+			    "resizing from %lu to %lu is not allowed",
+			    zinfo->main_zv->zv_name,
```
Comment (vishnuitta) [inline src/mgmt_conn.c]: why don't you just do zinfo->name?
and, put this log after drop_refcnt

Comment (mittachaitu) [reply src/mgmt_conn.c]: good to have drop_refcnt after logging since we are using `ZVOL_VOLUME_SIZE(zinfo->main_zv)` to get the size for logging.

Comment (mittachaitu) [reply src/mgmt_conn.c]: > why don't you just do zinfo->name?

Not aware of that I will update it

Comment (mittachaitu) [reply src/mgmt_conn.c]: Done both the changes

File: src/mgmt_conn.c [hunk 3]
```diff
@@ -1588,6 +1589,15 @@ process_message(uzfs_mgmt_conn_t *conn)
 			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED, hdrp);
 			break;
 		}
+		vol_size = ZVOL_VOLUME_SIZE(zinfo->main_zv);
+		if (resize_data->size < vol_size) {
+			uzfs_zinfo_drop_refcnt(zinfo);
```
Comment (vishnuitta) [inline src/mgmt_conn.c]: sorry for bad review last time..
as we are printing zinfo->name, it need to be done with refcount.. cc: @mittachaitu

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**